app.js

```diff
@@ -0,0 +1,112 @@
+var restler = require('restler');
+var uuid = require('node-uuid');
+var sslRootCAs = require('ssl-root-cas/latest');
+sslRootCAs.inject();
+var fs = require('fs');
+var request = require('request');
+var async = require('async');
+var express = require('express');
+var app = express();
+var multer = require('multer');
+var upload = multer({dest:'./uploads/'});
+var exec = require('child_process').exec;
+var cmd = '/usr/games/fortune | /usr/games/cowsay';
+var Constants = require(__dirname + '/Constants.js')
+
+var stlFilePath = "./3dFiles/original.stl";
+var myRes = null;
+
+app.get('/', function(req, res) {
+    exec(cmd, function(error, stdout, stderr) {
+        console.log('Running fortune into cowsay');
+        console.log(stdout);
+        console.log(stderr);
+        if (error != null) console.log(error);
+    }).stdout.pipe(res);
+});
+
+app.post('/image', upload.single('shapeJS_img'), function(req, res) {
+    var uuidJS = uuid.v4();
+    myRes = res;
+    // console.log(req.file);
+
+    var options = {
+        multipart: true,
+        headers: {},
+        data: {
+            shapeJS_img: restler.file('./uploads/' + req.file.filename, null, req.file.size),
+            jobID: uuidJS,
+            script: Constants.JS_2D_TO_3D
+        }
+    }
+    restler.post(Constants.UPDATE_SCENE_ENDPOINT, options).on('complete', function(response, body) {
+        console.log();
+        console.log(response);
+        console.log();
+        //console.log(error);
+        // console.log(body);
+
+        get3DModel(uuidJS, body);
+    });
+});
+
+function get3DModel(uuidJS, body) {
+    restler.get(Constants.SAVE_MODEL_CACHED_ENDPOINT + uuidJS, {
+        decoding: 'buffer'
+    }).on('complete', function(body) {
+        fs.writeFile(stlFilePath, body, function(err) {
+            if (err) {
```

Comment:
If you don't pass the response object between these functions, it's not possible to close off the response with a proper error message in the event of an error.

If you pass it `res`, then you should be able to do something like this:

```js
function get3DModel(res, uuidJS, body) {
  // ...
  fs.writeFile(stlFilePath, body, function(err) {
    if (err) { 
      res.status(500).end('There was a problem converting your file to STL'); 
    } else {
      runFbxConv(res);
    }
}
```